wsd/AdminModel.cpp

```diff
@@ -343,17 +343,17 @@ size_t AdminModel::getKitsJiffies() const
     size_t totalJ = 0;
     for (auto& it : _documents)
     {
-        if (!it.second->isExpired())
+        if (!it.second.isExpired())
         {
-            const int pid = it.second->getPid();
+            const int pid = it.second.getPid();
             if (pid > 0)
             {
                 unsigned newJ = Util::getCpuUsage(pid);
-                unsigned prevJ = it.second->getLastJiffies();
+                unsigned prevJ = it.second.getLastJiffies();
                 if(newJ >= prevJ)
                 {
                     totalJ += (newJ - prevJ);
-                    it.second->setLastJiffies(newJ);
+                    const_cast<Document&>(it.second).setLastJiffies(newJ);
```

Comment:
Odd that we need the `const_cast`.

Reply:
Its because of the "for (const auto& it : _documents)", which itself is needed because of the const of the method. 

It wasn't a problem when .second used to be a const std::shared_ptr<Document>& and the constness of the shared_ptr itself doesn't affect the contents. As in, this is allowed:

#include <memory>
int main(void)
{
  const std::shared_ptr<int> foo = std::make_shared<int>(42);
  *foo = 101;
  return *foo;
}